.github/workflows/ci.yml

```
@@ -0,0 +1,292 @@
+      run: cargo fmt -- --check
+      
+    # Step 8: Run clippy for code quality and style checks
+    - name: Run clippy
+      working-directory: ./backend
+      run: cargo clippy -- -D warnings
+      
+    # Step 9: Security audit for vulnerabilities
+    - name: Security audit
+      working-directory: ./backend
+      run: cargo audit || echo "Security audit failed - check dependencies"
+      
```

YAMLlint (1.37.1)

[error] trailing spaces

(trailing-spaces)